spring-vault-core/src/main/java/org/springframework/vault/authentication/AppRoleAuthentication.java

```diff
@@ -273,13 +274,13 @@ private String getRoleId(RoleId roleId) {
 				return (String) response.getRequiredData().get("role_id");
 			}
 			catch (HttpStatusCodeException e) {
-				throw new VaultException(String.format(
+				throw new VaultHttpException(String.format(
 						"Cannot unwrap Role id using AppRole: %s",
-						VaultResponses.getError(e.getResponseBodyAsString())));
+						VaultResponses.getError(e.getResponseBodyAsString())), e.getStatusCode());
 			}
 		}
 
-		throw new IllegalArgumentException("Unknown RoleId configuration: " + roleId);
+		throw new VaultClientException("Unknown RoleId configuration: " + roleId);
```

Comment:
Reasonable. Upon looking this would mean things were whacked. Another option would be subclasses for each, and throw in the factory, but this probably fine. So agreed!